python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -139,6 +150,7 @@ def _get_table_slice(
         schema: str
         table: str
         partition_value: Optional[Union[TimeWindow, str]] = None
+        partitions: List[TablePartition] = []
```

Comment:
It feels confusing to me to represent this as multiple `TablePartition` objects - logically, this slice is operating on a single partition, not multiple partitions.

It seems like we should either:
- Allow `TablePartition` to contain a multi-partition.
- Rename `TablePartition` to something that communicates that it's a particular partition dimension.

Reply:
ah interesting. I prefer renaming to `PartitionDimension` vs nesting another object within TablePartition. will do

Reply:
Could we rename to `TablePartitionDimension`? `PartitionDimension` seems to conflict a bit with `PartitionDimensionDefinition` which is a known entity